server/src/main/java/org/elasticsearch/index/mapper/flattened/FlattenedFieldMapper.java

```diff
@@ -359,7 +370,7 @@ public BytesRef indexedValueForSearch(Object value) {
         @Override
         public IndexFieldData.Builder fielddataBuilder(String fullyQualifiedIndexName, Supplier<SearchLookup> searchLookup) {
             failIfNoDocValues();
-            return new KeyedFlattenedFieldData.Builder(name(), key);
+            return new KeyedFlattenedFieldData.Builder(name(), key, toScriptField);
```

Comment:
Since this is the top level, I think we can continue to use the same design as elsewhere where we use a method ref here. Since KeyedFlattenedFieldType is different from RootFlattenedFieldType I think this should determine its own, and same with RootFlattenedFieldType.

Reply:
Done.